haystack/nodes/other/shaper.py

```diff
@@ -9,66 +12,92 @@
 logger = logging.getLogger(__name__)
 
 
-def rename(value: Any) -> Tuple[Any]:
+def rename(value: Any) -> Any:
     """
     Identity function. Can be used to rename values in the invocation context without changing them.
 
     Example:
 
     ```python
-    assert rename(1) == (1, )
+    assert rename(1) == 1
     ```
     """
-    return (value,)
+    return value
 
 
-def value_to_list(value: Any, target_list: List[Any]) -> Tuple[List[Any]]:
+def value_to_list(value: Any, target_list: List[Any]) -> List[Any]:
     """
     Transforms a value into a list containing this value as many times as the length of the target list.
 
     Example:
 
     ```python
-    assert value_to_list(value=1, target_list=list(range(5))) == ([1, 1, 1, 1, 1], )
+    assert value_to_list(value=1, target_list=list(range(5))) == [1, 1, 1, 1, 1]
     ```
     """
-    return ([value] * len(target_list),)
+    return [value] * len(target_list)
 
 
-def join_lists(lists: List[List[Any]]) -> Tuple[List[Any]]:
+def join_lists(lists: List[List[Any]]) -> List[Any]:
     """
     Joins the passed lists into a single one.
 
     Example:
 
     ```python
-    assert join_lists(lists=[[1, 2, 3], [4, 5]]) == ([1, 2, 3, 4, 5], )
+    assert join_lists(lists=[[1, 2, 3], [4, 5]]) == [1, 2, 3, 4, 5]
     ```
     """
     merged_list = []
     for inner_list in lists:
         merged_list += inner_list
-    return (merged_list,)
+    return merged_list
 
 
-def join_strings(strings: List[str], delimiter: str = " ") -> Tuple[str]:
+def join_strings(strings: List[str], delimiter: str = " ", str_replace: Optional[Dict[str, str]] = None) -> str:
     """
     Transforms a list of strings into a single string. The content of this string
     is the content of all original strings separated by the delimiter you specify.
 
     Example:
 
     ```python
-    assert join_strings(strings=["first", "second", "third"], delimiter=" - ") == ("first - second - third", )
+    assert join_strings(strings=["first", "second", "third"], delimiter=" - ", str_replace={"r": "R"}) == "fiRst - second - thiRd"
     ```
     """
-    return (delimiter.join(strings),)
+    str_replace = str_replace or {}
+    return delimiter.join([format_string(string, str_replace) for string in strings])
 
 
-def join_documents(documents: List[Document], delimiter: str = " ") -> Tuple[List[Document]]:
+def format_string(string: str, str_replace: Optional[Dict[str, str]] = None) -> str:
+    """
+    Transforms a string using a substitution dict.
+
+    Example:
+
+    ```python
+    assert format_string(string="first", str_replace={"r": "R"}) == "fiRst"
+    ```
+    """
+    str_replace = str_replace or {}
+    return reduce(lambda s, kv: s.replace(*kv), str_replace.items(), string)
+
+
+def join_documents(
```

Comment:
Used as `join` in PromptTemplate syntax

Reply:
Really? Isn't `join_documents_to_string` used in the PromptTemplate? :thinking:

Reply:
No, we expose this function as join in the globals of `eval`. No hard feelings on that, I thought `join` is simpler and easier to understand.